File: Patterns/Abstract Factory Pattern/Cpp/Pizza/Ingredient/NYPizzaIngredientFactory.cpp
#include "./PizzaIngredientFactory.cpp"

#include "./Dough/ThinCrustDough.cpp"
#include "./Cheese/ReggianoCheese.cpp"
#include "./Sauce/MarinaraSauce.cpp"
#include "./Pepperoni/SlicedPepperoni.cpp"
#include "./Clams/FreshClams.cpp"
#include "./Veggies/Veggies.cpp"
#include "./Veggies/Garlic.cpp"
#include "./Veggies/Onion.cpp"
#include "./Veggies/Mushroom.cpp"
#include "./Veggies/RedPepper.cpp"

class NYPizzaIngredientFactory : public PizzaIngredientFactory {
public:
    Dough* createDough() override {
        return new ThinCrustDough();
    }

    Sauce* createSauce() override {
        return new MarinaraSauce();
    }

    Cheese* createCheese() override {
        return new ReggianoCheese();
    }

    Pepperoni* createPepperoni() override {
        return new SlicedPepperoni();
    }

    Clams* createClams() override {
        return new FreshClams();
    }

    vector<Veggies*> createVeggies() override {
        return { new Garlic(),new Onion(),new Mushroom(),new RedPepper() };
    }
};

File: Patterns/Abstract Factory Pattern/CSharp/Pizza/Ingredient/NYPizzaIngredientFactory.cs
namespace Pizza.Ingredient
{
    public class NYPizzaIngredientFactory : PizzaIngredientFactory
    {
        public Dough.Dough CreateDough() => new Dough.ThinCrustDough();
        public Sauce.Sauce CreateSauce() => new Sauce.MarinaraSauce();
        public Cheese.Cheese CreateCheese() => new Cheese.ReggianoCheese();
        public Pepperoni.Pepperoni CreatePepperoni() => new Pepperoni.SlicedPepperoni();
        public Clams.Clams CreateClams() => new Clams.FreshClams();
        public Veggies.Veggies[] CreateVeggies() => 
            new Veggies.Veggies[] { new Veggies.Garlic(),new Veggies.Mushroom(),new Veggies.Onion(),new Veggies.RedPepper() };
    }
}